File: demo_RES.m


%% Low level of independent noise
simOpts = struct('noiseVar', 0.16, 'noiseCorrCoeff', 0, 'sourceEnergy', 6.3);

%% Medium level of correlated noise 
%simOpts = struct('noiseVar', 4, 'noiseCorrCoeff', 0.5, 'sourceEnergy', 6.3);
%
%% High level of highly correlated noise
%simOpts = struct('noiseVar', 16, 'noiseCorrCoeff', 0.8, 'sourceEnergy', 6.3);

data    = rotating_energy_sources(70, simOpts);

dirname   = strcat( 'RES-rn' );
bfilename = strcat( dirname, '/res' );
mkdir( dirname );

myOpts = struct( 'cmethod', 'phase', ...
	'boolParfor', false, ... 
	'boolUseSavedData', 0, ...
	'errorRate', 0.1, ...
	'bfilename', bfilename);


tic;
Z = pasf(data, 2, myOpts);
toc;

[d1, d2, d3, d4] = size(Z);
rng = prctile( Z(:), [1 99] ) + [-eps eps];
for f = 1:d3,
	frame = Z(:,:,f,1);
	for c = 2:d4,
		frame = horzcat(frame, Z(:, :, f, c) );
	end
	imagesc( frame, rng);  
	title('Component #1, Component #2, Remainder, and Input Signal');
	daspect([1 1 1]);
	axis off;
	pause(0.5);
end

File: demo_RES.py
import time
import numpy as np
import matplotlib.pyplot as plt
import matlab.engine

eng = matlab.engine.start_matlab()

simOpts = eng.struct('noiseVar', 0.16, 'noiseCorrCoeff', 0,
        'sourceEnergy', 6.3)
data = eng.rotating_energy_sources(70, simOpts)


dirname   = 'RES-rn'
bfilename = dirname + '/res'
eng.mkdir( dirname )

myOpts = eng.struct( 'cmethod', 'phase',
	'boolParfor', 0,
	'boolUseSavedData', 0,
	'errorRate', 0.1,
	'bfilename', bfilename)


eng.tic;
Components, Clusters, ClusterInfo, SDFInfo = eng.pasf(data, 2, myOpts, nargout=4)
eng.toc;

Components = np.array(Components._data).reshape(Components.size, order='F')
print( Components.shape )
d1, d2, d3, d4 = Components.shape

for i in range(d3):
    fig = Components[:,:,i,0]
    for c in range(1, d4):
        fig = np.hstack((fig, Components[:,:,i,c]))

    plt.imshow(fig)
    plt.show(block=False)
    time.sleep(0.5)
    plt.close()

